OJS.Workers.ExecutionStrategies/RunSpaAndExecuteMochaTestsExecutionStrategy.cs

```diff
@@ -237,13 +290,49 @@ protected override IExecutor CreateExecutor()
             => this.CreateExecutor(ProcessExecutorType.Standard);
 
         private ICollection<TestResult> RunIndividualTest(
-            string codeSavePath,
+            string preExecuteCodeSavePath,
             IExecutor executor,
             IExecutionContext<TestsInputModel> executionContext,
             TestContext test)
         {
-            var processExecutionResult = this.Execute(executionContext, executor, codeSavePath, test.Input);
-            return this.ExtractTestResultsFromReceivedOutput(processExecutionResult.ReceivedOutput, test.Id);
+            var preExecutionResult = this.Execute(executionContext, executor, preExecuteCodeSavePath, test.Input);
+            var match = Regex.Match(preExecutionResult.ReceivedOutput, @"Container port: (\d+);Container name: ([a-zA-Z-_]+);");
+            if (match.Success)
+            {
+                this.PortNumber = int.Parse(match.Groups[1].Value);
+                var containerName = match.Groups[2].Value;
+
+                var filePath = FileHelpers.BuildPath(this.TestsPath, $"{test.Id}{JavaScriptFileExtension}");
+
+                // pass in container name in order to close container after execution
+                // pass test file path to mocha so it executes only this test file, and not all test files each run
+                var processedPythonCodeTemplate = this.PythonCodeTemplate
+                    .Replace(ContainerNamePlaceholder, containerName)
+                    .Replace(TestFilePathPlaceholder, filePath);
+
+                var mainCodeSavePath = this.SavePythonCodeTemplateToTempFile(processedPythonCodeTemplate);
+
+                this.SaveTestsToFiles(executionContext.Input.Tests);
+
+                var processExecutionResult = this.Execute(executionContext, executor, mainCodeSavePath, test.Input);
+                return this.ExtractTestResultsFromReceivedOutput(processExecutionResult.ReceivedOutput, test.Id);
+            }
+            else
+            {
+                return new List<TestResult>()
+                {
+                 new TestResult
+                    {
+                        Id = test.Id,
```

Comment:
Isnt this going to show Test#: 123456 on the front end instead of the text index?

Reply:
NVM this is a separate issue https://github.com/SoftUni-Internal/exam-systems-issues/issues/570